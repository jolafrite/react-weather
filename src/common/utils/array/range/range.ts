export const range = (from: number, to: number, step = 1): number[] => {
  return Array.from(
    { length: Math.floor((to - from) / step) + 1 }, (v, k) => from + k * step
  );
};
