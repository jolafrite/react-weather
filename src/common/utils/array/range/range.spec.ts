import { range } from "./range";

describe('range', () => {

  test('range 1 to 10', () => {
    expect(
      range(1, 10)
    ).toMatchSnapshot();
  });

  test('delta 2', () => {
    expect(
      range(1, 10, 2)
    ).toMatchSnapshot();
  });

});
