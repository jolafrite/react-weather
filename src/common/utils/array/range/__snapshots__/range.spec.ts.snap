// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`range delta 2 1`] = `
Array [
  1,
  3,
  5,
  7,
  9,
]
`;

exports[`range range 1 to 10 1`] = `
Array [
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
]
`;
